### PR TITLE
Adding module to manage RabbitMQ connections and channels

### PR DESCRIPTION
**Pull request by gausnes**
Initial pass at adding a RabbitMQ module, a considerable amount of this functionality was inspired the module written for ticker. The primary difference is changing how the HealthCheck works.

In general this library is intended to encapsulate the connection to RabbitMQ and manage that outside of the service. The maximum default channels per connection is 65535 which in my opinion outside normal use case for RabbitMQ interaction.

Consumers of this library are responsible for managing the respective channels they create and should implement correct shutdown procedure of those channels. We might want to add this functionality to manage channels to this library to ensure that channels are being closed cleanly.

\cc @aaron-miller @rappleg @darksun @elventear @charliek

### CODE REVIEW

File: dropwizard-rabbitmq/build.gradle [hunk 1]
```diff
@@ -0,0 +1,11 @@
+dependencies {
+    compile project(':dropwizard-guice')
```
Comment (aaron-miller) [inline dropwizard-rabbitmq/build.gradle]: Do we want to specify a version here?

File: dropwizard-rabbitmq/src/main/java/smartthings/dw/rabbitmq/RabbitMQConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @NotEmpty
+    private String virtualHost = "/";
+
+    @NotEmpty
+    private String username;
+
+    @NotEmpty
+    private String password;
+
+    private Integer port = null;
```
Comment (aaron-miller) [inline dropwizard-rabbitmq/src/main/java/smartthings/dw/rabbitmq/RabbitMQConfiguration.java]: Should we default this to the default rabbitmq port?

Comment (gausnes) [reply dropwizard-rabbitmq/src/main/java/smartthings/dw/rabbitmq/RabbitMQConfiguration.java]: I don't have a strong preference here, if the port isn't set in the ConnectionFactory it does to the default, see `RabbitMQConfigurationSpec`. If we want to more specific about the port I would be fine with that.

File: dropwizard-rabbitmq/build.gradle (hunk 1 above)
Comment (gausnes) [reply dropwizard-rabbitmq/build.gradle]: I was just following what the other modules are doing, not sure if there is reasoning behind not specifying a version

Comment (elventear) [reply dropwizard-rabbitmq/build.gradle]: `dropwizard-guice` is another module in `dropwizard-common`, so you want to use always the latest.

**Pull request merged**